doc/releases/release-notes-2.6.rst

```diff
@@ -29,6 +29,9 @@ interface and listing all issues with the `bug label
 API Changes
 ***********
 
+* Driver APIs now return ``-ENOSYS`` if optional functions are not implemented.
+  If the feature is not supported by the hardware ``-ENOTSUP`` will be returned.
```

Comment:
This should note that formerly `-ENOTSUP` was returned for both failure modes, since the change may require existing code that tests only for that value to be changed.